codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -398,6 +398,12 @@ private void verifyContinuationFrame() throws Http2Exception {
         }
     }
 
+    private void verifyUnknownFrame() throws Http2Exception {
+        if (headersContinuation != null) {
+            throw connectionError(PROTOCOL_ERROR, "Extension frames must not be in the middle of headers.");
```

Comment:
OK, that make sense!
Though violate the rule will cause connection, provide more information about stream id will help troubleshooting or something.